app/shared/app-data/src/androidMain/kotlin/domain/torrent/service/proxy/TorrentFileEntryProxy.kt

```diff
@@ -0,0 +1,91 @@
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import me.him188.ani.app.domain.torrent.IDisposableHandle
+import me.him188.ani.app.domain.torrent.IRemotePieceList
+import me.him188.ani.app.domain.torrent.IRemoteTorrentFileEntry
+import me.him188.ani.app.domain.torrent.IRemoteTorrentFileHandle
+import me.him188.ani.app.domain.torrent.ITorrentFileEntryStatsCallback
+import me.him188.ani.app.domain.torrent.parcel.PTorrentFileEntryStats
+import me.him188.ani.app.domain.torrent.parcel.PTorrentInputParameter
+import me.him188.ani.app.torrent.anitorrent.session.AnitorrentDownloadSession
+import me.him188.ani.app.torrent.api.files.TorrentFileEntry
+import me.him188.ani.utils.coroutines.childScope
+import me.him188.ani.utils.io.absolutePath
+import kotlin.coroutines.CoroutineContext
+
+class TorrentFileEntryProxy(
+    private val delegate: TorrentFileEntry,
+    context: CoroutineContext
+) : IRemoteTorrentFileEntry.Stub(), CoroutineScope by context.childScope() {
+    override fun getFileStats(flow: ITorrentFileEntryStatsCallback?): IDisposableHandle {
+        val job = launch {
+            delegate.fileStats.collect {
+                flow?.onEmit(PTorrentFileEntryStats(it.downloadedBytes, it.downloadProgress))
+            }
+        }
+
+        return DisposableHandleProxy { job.cancel() }
+    }
+
+    override fun getLength(): Long {
+        return delegate.length
+    }
+
+    override fun getPathInTorrent(): String {
+        return delegate.pathInTorrent
+    }
+
+    override fun getPieces(): IRemotePieceList {
+        return PieceListProxy(delegate.pieces, coroutineContext)
+    }
+
+    override fun getSupportsStreaming(): Boolean {
+        return delegate.supportsStreaming
+    }
+
+    override fun createHandle(): IRemoteTorrentFileHandle {
+        return TorrentFileHandleProxy(delegate.createHandle(), coroutineContext)
+    }
+
+    override fun resolveFile(): String {
+        return runBlocking { delegate.resolveFile().absolutePath }
+    }
+
+    override fun resolveFileMaybeEmptyOrNull(): String? {
+        return delegate.resolveFileMaybeEmptyOrNull()?.absolutePath
+    }
+
+    override fun getTorrentInputParams(): PTorrentInputParameter? {
+        require(delegate is AnitorrentDownloadSession.AnitorrentEntry) {
+            "Expected delegate instance is AnitorrentEntry, actual $delegate"
+        }
+        val torrentInputParameters = runBlocking { delegate.createTorrentInputParameters() }
+
+        return PTorrentInputParameter(
+            file = torrentInputParameters.file.absolutePath,
+            logicalStartOffset = torrentInputParameters.logicalStartOffset,
+            bufferSize = torrentInputParameters.bufferSize,
+            size = torrentInputParameters.size,
+        )
+    }
+
+    override fun torrentInputOnWait(pieceIndex: Int) {
+        require(delegate is AnitorrentDownloadSession.AnitorrentEntry) {
```

Comment:
```suggestion
        check(delegate is AnitorrentDownloadSession.AnitorrentEntry) {
```
这个情况适用 check

require 只检查函数的直接参数, 例如 pieceIndex